File: pset6/mario_less.py
##
# Calculates and draws a single "mario" pyramide
##

import cs50


def main():
    while True:
        print("Height: ", end="")
        i = cs50.get_int()
        if input_okay(i):
            break
    draw_pyramid(i)
    exit(0)


def input_okay(i):
    if i < 0 or i > 23:
        return False
    return True


def draw_pyramid(rows):
    cells = rows + 1
    hashes = 2
    for k in range(rows):
        draw_symbols(" ", cells - hashes)
        draw_symbols("#", hashes)
        print()
        hashes += 1


def draw_symbols(s, c):
    for i in range(c):
        print(s, end="")

if __name__ == "__main__":
    main()

File: pset1/mario_less.c
#include <stdio.h>
#include <cs50.h>

bool input_okay(int i);
void draw_pyramid(int i);
void draw_symbols (char c, int count);

int main(void)
{
    int i;
    do
    {
        printf("Height: ");
        i = get_int();
    }
    while (!input_okay(i));
    draw_pyramid(i);
}

bool input_okay(int i)
{
    if ( (i<0) || (i>23) ) {
        return false;
    }
    return true;
}

void draw_pyramid(int i)
{
    int rows    = i;
    int cells   = i +1;
    int hashes  = 2;
    for (int n = 0; n < rows; n++) 
    {
        draw_symbols(' ',cells - hashes);
        draw_symbols('#',hashes);
        printf("\n");
        hashes++;
    }
}

void draw_symbols (char c, int count)
{
    for (int i = 0; i < count; i ++)
    {
        printf("%c",c);
    }
}
